### ISSUE TITLE
Update Facebook Login to latest Code/API

### ISSUE THREAD

**Issue by WOs8lfTkz3E5g7I4Lm** [geodesicsolutions-community/geocore-community#61]
I am unable to get it to work with v20.0.0 Beta-2. 

The first time you try to connect it goes through the motions and Facebook appears to work correctly and return to the proper callback URL:

index.php?a=4&code=[...]

Also, the "Login with Facebook" button in the header and the operates differently than the same button that appears on the login page.

The button in the header pretty much does nothing and you end up in the same place.

The button on the Login Form actually takes you to FB where you get an error:

`Sorry, something went wrong.

We're working on getting this fixed as soon as we can.`

**Comment by vicos59** [comment]
Current version uses Facebook PHP SDK (v.3.2.3)

https://github.com/facebookarchive/facebook-php-sdk

This is now deprecated. New version is v5.7:

https://github.com/facebookarchive/php-graph-sdk

But, the bad news is the repository has been archived by the owner. It is now read-only. Last update was 3 years. There is at least one recent issue indicating that there is a deprecation issue with PHP 8.1.

No indication in the readme about a recommended replacement.

**Comment by vicos59** [comment]
Some folks rightfully complaining about how Facebook like to abandon libraries without a replacement or support. Some solutions are also proposed:

https://github.com/facebookarchive/php-graph-sdk/issues/1217

One gent said he is using the following and the migration was painless (but, assuming from a recent version, not the old v3):

https://github.com/thephpleague/oauth2-facebook

They say they support PHP 7.3, 7.4, 8.0. Updates are relatively recent.

Another gent says, "I created a refactored project for PHP 8.0+."

https://github.com/janu-software/facebook-php-sdk

0.1 is the one version available and he indicates support up tp PHP 8.0. Last updates with recent days.

**Comment by jonyo** [comment]
Yikes, yeah looks like they are abandoning the old libraries.  I hadn't looked into it further yet but was hoping there was something they would be replaced by.

This is looking like a pretty big rewrite to get it working.

Another option, IF we have to rewrite it anyways, is to integrate [Firebase auth](https://firebase.google.com/docs/auth) - then we would have Facebook, Google, Microsoft, Apple logins as well...  Plus they have phone login option, though that would probably take even more work since the only info we have without asking for more, would be the phone number.

I wonder if any site owners would be interested in sponsoring development for this?  I have experience with Firebase, but maybe this would be a good chance to get another developer "in the fold" so to speak.  If we can find site owners wishing to pay to have the feature added, then we could reach out to a few developers.

**Comment by jonyo** [comment]
Info on firebase auth: https://youtu.be/8sGY55yxicA

**Comment by vicos59** [comment]
From my personal perspective, I'm hanging on by a thread. I need a stable version with PayPal Pro and Facebook Login functioning so I can move off the old server, which is ready to go up in smoke. I know its not your problem, but my motivation would be to just get FB working and do the rest as a later enhancement.

**Comment by jonyo** [comment]
@vicos59 understood, I've done a quick fix for the paypal pro as mentioned in another issue.  And I'll look into whether a quick fix is possible for the facebook login as well.

Like a year ago or so we did do some changes, you just have to add certain URL's to the config to be valid redirect URLs.  I think it is `https://example.com/index.php?a=4` for example (swap out your own domain).

When I have some time I'll try testing to see if that's all we need for facebook login or not, with luck maybe it is just a config issue.

**Comment by D0nTlOokB4cK** [comment]
im having an Xcode swift Facebook login button error
<img width="334" alt="Schermafbeelding 2022-03-05 om 09 46 31" src="https://user-images.githubusercontent.com/85488444/156875877-41c4e26a-2060-4755-9200-49eff5438552.png
<img width="839" alt="Schermafbeelding 2022-03-05 om 09 46 40" src="https://user-images.githubusercontent.com/85488444/156875908-8ffd3c82-1085-4be9-84a1-81db5bf89869.png">
">

**Comment by vicos59** [comment]
> im having an Xcode swift Facebook login button error <img alt="Schermafbeelding 2022-03-05 om 09 46 31" width="334" src="https://user-images.githubusercontent.com/85488444/156875877-41c4e26a-2060-4755-9200-49eff5438552.png">
> 
> <img alt="Schermafbeelding 2022-03-05 om 09 46 40" width="839" src="https://user-images.githubusercontent.com/85488444/156875908-8ffd3c82-1085-4be9-84a1-81db5bf89869.png">
> 
> have any one faced this issue?

Yes, GeoCore v18.02 and GeoCoreCE v20.0.0 no longer work with Facebook. Facebook has upgraded their API several times over the past few years and their PHP API on github is out of date and no longer supported by them.

This issue is to get it working again.

**Comment by vicos59** [comment]
> Another option, IF we have to rewrite it anyways, is to integrate [Firebase auth](https://firebase.google.com/docs/auth) - then we would have Facebook, Google, Microsoft, Apple logins as well... Plus they have phone login option, though that would probably take even more work since the only info we have without asking for more, would be the phone number.
> 
> I wonder if any site owners would be interested in sponsoring development for this? I have experience with Firebase, but maybe this would be a good chance to get another developer "in the fold" so to speak. If we can find site owners wishing to pay to have the feature added, then we could reach out to a few developers.

It would be nice to have FB, Google and Twitter login capabilities. Personally, I always use Google.

Once we get the official first release out, I can pitch this the FB page and see if we can get people interested.

My only concerns with using a Google product are that typically Google only offers something if:

- They want to own your data
- They want to own you
- They want to offer it for free at first and over the years funnel you in to no other option but to pay to use it.

**Comment by D0nTlOokB4cK** [comment]
oké thank you for your support. im using firebase but still got this error...

**Comment by D0nTlOokB4cK** [comment]
so they don't support the version anymore even with firebase auth? because many I have written some error in the code

**Comment by vicos59** [comment]
@jonyo 

I pulled the new libraries to have a look:

https://github.com/thephpleague/oauth2-facebook
https://github.com/janu-software/facebook-php-sdk

They are both way completely different than the way things are done now. I'm afraid that this one is above my current pay grade.

The only advice I can offer is that oauth2-facebook has way more forks and people watching. So, it might be the better horse to bet on as far as longevity.

I also took a look at Firebase auth. Looks like it is mainly javascript, which is also not one of my strong points.

**Comment by vicos59** [comment]
> 

I may have misunderstood you. I was referring to Facebook support in the GeoCore product, named Social COnnect. GeoCore does not currently use Firebase auth. Sounds like you are referring to another product?

**Comment by D0nTlOokB4cK** [comment]
i am currently using Xcode: Swift to build an app.
currently im using firebase auth but I get this error. 

so I thought maybe I written something wrong in my code. im a learner nd I have this from the internet but I retyped the code for 3 times and still gives the same error.

**Comment by vicos59** [comment]
I understand now. So, basically you asked in the wrong place. Perhaps this is a good place to look for help:

https://firebase.google.com/community

**Comment by vicos59** [comment]
> Like a year ago or so we did do some changes, you just have to add certain URL's to the Facebook "app" config to be valid redirect URLs. I think it is https://example.com/index.php?a=4 for example (swap out your own domain).

OK, I already had the redirect URL set properly.

![fb-login](https://user-images.githubusercontent.com/6634041/157099073-3aaa55d1-f69c-44b7-8a7b-a0e514d8031f.JPG)

Be nice if FB offered some debug data somewhere...or do they....

**Comment by jonyo** [comment]
So I refreshed my memory...  It depends on what you have set in the settings for the social connect addon.  In the settings, you'll see `Valid OAuth Redirect URI:` - that is what you need to have there.  I think the default is the a=4 url but it depends if you changed the settings for where to go after you log in.

**Comment by vicos59** [comment]
> So I refreshed my memory... It depends on what you have set in the settings for the social connect addon. In the settings, you'll see `Valid OAuth Redirect URI:` - that is what you need to have there. I think the default is the a=4 url but it depends if you changed the settings for where to go after you log in.

Yep, I saw that and that is what I pasted into FB back when I first set it up in my Developer account.
